src/oss/langchain-test.mdx

```diff
@@ -1,3 +1,1057 @@
+    """Your agent function that returns trajectory messages."""
+    return agent.invoke(inputs)["messages"]
+
+experiment_results = client.evaluate(
+    run_agent,
+    data="your_dataset_name",
+    evaluators=[trajectory_evaluator]
+)
+```
+
+Results will be automatically logged to LangSmith.
+
+
+</Accordion>
+
+<Tip>
+To learn more about evaluating your agent, see the [LangSmith docs](/langsmith/pytest).
+</Tip>
+:::
+
+:::js
+LangSmith offers two main approaches for running evaluations: [Vitest/Jest](/langsmith/vitest-jest) integration and the `evaluate` function.
+
+### Using Vitest/Jest integration
+
+```typescript
+import * as ls from "langsmith/vitest";
+// import * as ls from "langsmith/jest";
+
+import { createTrajectoryLLMAsJudge, TRAJECTORY_ACCURACY_PROMPT } from "agentevals";
+
+const trajectoryEvaluator = createTrajectoryLLMAsJudge({
+  model: "openai:o3-mini",
+  prompt: TRAJECTORY_ACCURACY_PROMPT,
+});
+
+ls.describe("trajectory accuracy", () => {
+  ls.test("accurate trajectory", {
+    inputs: {
+      messages: [
+        {
+          role: "user",
+          content: "What is the weather in SF?"
+        }
+      ]
+    },
+    referenceOutputs: {
+      messages: [
+        new HumanMessage("What is the weather in SF?"),
+        new AIMessage({
+          content: "",
+          tool_calls: [
+            { id: "call_1", name: "get_weather", args: { city: "SF" } }
+          ]
+        }),
+        new ToolMessage({
+          content: "It's 75 degrees and sunny in SF.",
+          tool_call_id: "call_1"
+        }),
+        new AIMessage("The weather in SF is 75 degrees and sunny."),
+      ],
+    },
+  }, async ({ inputs, referenceOutputs }) => {
+    const result = await agent.invoke({
+      messages: [new HumanMessage("What is the weather in SF?")]
+    });
+    
+    ls.logOutputs({ messages: result.messages });
+
+    await trajectoryEvaluator({
+      inputs,
+      outputs: result.messages,
+      referenceOutputs,
+    });
+  });
+});
+```
+
+Run the evaluation with your test runner:
+
+```bash
+vitest run test_trajectory.eval.ts
+# or
+jest test_trajectory.eval.ts
+```
+
+### Using the evaluate function
+
+Alternatively, you can create a dataset in LangSmith and use the `evaluate` function:
+
+```typescript
+import { evaluate } from "langsmith/evaluation";
+import { createTrajectoryLLMAsJudge, TRAJECTORY_ACCURACY_PROMPT } from "agentevals";
+
+const trajectoryEvaluator = createTrajectoryLLMAsJudge({
+  model: "openai:o3-mini",
+  prompt: TRAJECTORY_ACCURACY_PROMPT,
+});
+
+async function runAgent(inputs: any) {
+  const result = await agent.invoke(inputs);
+  return result.messages;
+}
+
+await evaluate(
+  runAgent,
+  {
+    data: "your_dataset_name",
+    evaluators: [trajectoryEvaluator],
+  }
+);
+```
+
+Results will be automatically logged to LangSmith.
+
+<Tip>
+To learn more about evaluating your agent with LangSmith, see the [LangSmith docs](/langsmith/vitest-jest).
+</Tip>
+:::
+
+## Caching
```

Comment:
What about naming it more explicitly: "## Recording & Replaying HTTP Calls"

Caching feels ambiguous to me, especially since we have other forms of caching (e.g., caching for embeddings models)